src/main/java/com/redhat/pantheon/asciidoctor/AsciidoctorPool.java

```diff
@@ -44,9 +44,14 @@ public AsciidoctorPool(@Reference GlobalConfig globalConfig) {
      */
     public Asciidoctor borrowObject(Resource base) {
         Asciidoctor asciidoctor = super.borrowObject();
-        asciidoctor.javaExtensionRegistry().includeProcessor(
-                new SlingResourceIncludeProcessor(base));
-        return asciidoctor;
+        try {
+            asciidoctor.javaExtensionRegistry().includeProcessor(
+                    new SlingResourceIncludeProcessor(base));
```

Comment:
I've always thought that we should change this. The responsibility of declaring the necessary extensions should be left up to the caller of the pool, and the pool simply clears them up upon return of the borrowed object.

Reply:
Ok, that's probably a good idea. I suspect that 95% of the time, the intention will be to build a Resource, but I suppose at some point in the future we might be interested in building some in-memory-only asciidoc. We could refactor this then. The immediate concern was that I wrote some code that caused the constructor of the SlingResourceIncludeProcessor to throw an exception, which had the ultimate effect of the asciidcotor instance never being returned to the pool - so after failing to render 10 docs, the system was totally unable to render any further docs. This code update fixes that issue.

Reply:
Yes, I saw that, which made be go down the thought process... but if we remove line 48 and leave it up to the caller to add their own asciidoctor extensions, we remove the chance for there to be an exception and hence the need for this method in the first place.